lib/Common/Memory/MarkContext.inl

```diff
@@ -197,6 +197,8 @@ void MarkContext::ProcessMark()
             // Process the previously retrieved entry.
             ScanObject<parallel, interior>(current.obj, current.byteCount);
 
+            _mm_prefetch((char *)*(next.obj), _MM_HINT_T0);
```

Comment:
The double indirection to get to the actual object is a costly hotspot during the processMark phase. Accessing the data pointed by **obj member of 'next' is incurring significant penailty. This prefetch  alleviates some of the latency by prefetching the first deref prior to the next loop iteration where it will be used.  This prefetch improves Octane.SplayLatency score by 10% on core.